File: storage/data_original/at_coder/arc013/A/4834192.py
d,e,f = map(int,input().split())
a,b,c = map(int,input().split())
q = (d//a) * (e//b) * (f//c)
w = (e//a) * (d//b) * (f//c)
r = (f//a) * (e//b) * (d//c)
t = (d//a) * (f//b) * (e//c)
y = (e//a) * (f//b) * (d//c)
u = (f//a) * (d//b) * (e//c)
print(max(q,w,r,t,y,u))

File: data/data_transpiled/at_coder/arc013/A/4834192.java
public static void d ( int d , int e , int f ) {
  int a = Integer . parseInt ( input . nextLine ( ) ) ;
  int b = Integer . parseInt ( input . nextLine ( ) ) ;
  int c = Integer . parseInt ( input . nextLine ( ) ) ;
  int q = ( d / a ) * ( e / b ) * ( f / c ) ;
  int w = ( e / a ) * ( d / b ) * ( f / c ) ;
  int r = ( f / a ) * ( e / b ) * ( d / c ) ;
  int t = ( d / a ) * ( f / b ) * ( d / c ) ;
  int y = ( e / a ) * ( f / b ) * ( d / c ) ;
  int u = ( f / a ) * ( d / b ) * ( e / c ) ;
  System . out . println ( max ( q , w , r , t , y , u ) ) ;
}
